diff -ru link-grammar-4.2.4/link-grammar/api.c link-grammar-4.2.4-fixed/link-grammar/api.c
--- link-grammar-4.2.4/link-grammar/api.c	2006-04-09 07:05:25.000000000 -0700
+++ link-grammar-4.2.4-fixed/link-grammar/api.c	2006-10-28 19:06:30.000000000 -0700
@@ -861,6 +861,10 @@
     return linkage;
 }
 
+int linkage_get_current_sublinkage(Linkage linkage) {
+    return( linkage->current );
+}
+
 int linkage_set_current_sublinkage(Linkage linkage, int index) {
     int errno;
     if ((index < 0) ||
Only in link-grammar-4.2.4/link-grammar: api.c.rej
Only in link-grammar-4.2.4-fixed/link-grammar: api.c~
diff -ru link-grammar-4.2.4/link-grammar/link-grammar.def link-grammar-4.2.4-fixed/link-grammar/link-grammar.def
--- link-grammar-4.2.4/link-grammar/link-grammar.def	2006-06-18 11:44:25.000000000 -0700
+++ link-grammar-4.2.4-fixed/link-grammar/link-grammar.def	2006-10-29 22:40:41.115153450 -0800
@@ -72,6 +72,7 @@
 sentence_get_nth_word
 sentence_nth_word_has_disjunction
 linkage_create
+linkage_get_current_sublinkage
 linkage_set_current_sublinkage
 linkage_delete
 linkage_get_sentence
@@ -114,3 +115,4 @@
 linkage_constituent_node_get_end
 lperrno
 lperrmsg
+string_delete
Only in link-grammar-4.2.4-fixed/link-grammar: link-grammar.def.orig
Only in link-grammar-4.2.4/link-grammar: link-grammar.def.rej
Only in link-grammar-4.2.4-fixed/link-grammar: link-grammar.def~
diff -ru link-grammar-4.2.4/link-grammar/link-includes.h link-grammar-4.2.4-fixed/link-grammar/link-includes.h
--- link-grammar-4.2.4/link-grammar/link-includes.h	2006-06-18 11:42:13.000000000 -0700
+++ link-grammar-4.2.4-fixed/link-grammar/link-includes.h	2006-10-29 22:40:48.008902530 -0800
@@ -219,6 +219,8 @@
 link_public_api(Linkage)
      linkage_create(int index, Sentence sent, Parse_Options opts);
 link_public_api(int)
+     linkage_get_current_sublinkage(Linkage linkage);
+link_public_api(int)
      linkage_set_current_sublinkage(Linkage linkage, int index);
 link_public_api(void)
      linkage_delete(Linkage linkage);
@@ -325,6 +327,15 @@
 extern link_public_api(char) 
      lperrmsg[];
 
+
+link_public_api(void)
+	string_delete(char *s);
+
+link_public_api(CNode *) 
+     linkage_constituent_tree(Linkage linkage);
+link_public_api(void)
+     linkage_free_constituent_tree(CNode * n);
+
 LINK_END_DECLS
 
 #endif
Only in link-grammar-4.2.4-fixed/link-grammar: link-includes.h.orig
Only in link-grammar-4.2.4/link-grammar: link-includes.h.rej
Only in link-grammar-4.2.4-fixed/link-grammar: link-includes.h~
