Only in link-grammar-4.2.2.patched/: autogen.err
diff -ubr link-grammar-4.2.2/link-grammar/constituents.h link-grammar-4.2.2.patched/link-grammar/constituents.h
--- link-grammar-4.2.2/link-grammar/constituents.h	2006-02-25 08:22:22.000000000 -0800
+++ link-grammar-4.2.2.patched/link-grammar/constituents.h	2006-07-01 15:17:35.000000000 -0700
@@ -15,15 +15,6 @@
 
 #include "link-includes.h"
 
-/* Invariant: Leaf if child==NULL */
-typedef struct CNode_s CNode;
-struct CNode_s {
-  char  * label;
-  CNode * child;
-  CNode * next;
-  int   start, end;
-};
-
 CNode * linkage_constituent_tree(Linkage linkage);
 void    linkage_free_constituent_tree(CNode * n);
 
Only in link-grammar-4.2.2/link-grammar: link-features.h
diff -ubr link-grammar-4.2.2/link-grammar/link-grammar.def link-grammar-4.2.2.patched/link-grammar/link-grammar.def
--- link-grammar-4.2.2/link-grammar/link-grammar.def	2006-04-04 11:01:36.000000000 -0700
+++ link-grammar-4.2.2.patched/link-grammar/link-grammar.def	2006-07-13 08:35:27.000000000 -0700
@@ -107,3 +107,6 @@
 issue_special_command
 lperrno
 lperrmsg
+linkage_constituent_tree
+linkage_free_constituent_tree
+string_delete
diff -ubr link-grammar-4.2.2/link-grammar/link-includes.h link-grammar-4.2.2.patched/link-grammar/link-includes.h
--- link-grammar-4.2.2/link-grammar/link-includes.h	2006-04-04 11:01:36.000000000 -0700
+++ link-grammar-4.2.2.patched/link-grammar/link-includes.h	2006-07-13 08:35:45.000000000 -0700
@@ -302,6 +302,24 @@
 extern link_public_api(char) 
      lperrmsg[];
 
+
+/* Invariant: Leaf if child==NULL */
+typedef struct CNode_s CNode;
+struct CNode_s {
+  char  * label;
+  CNode * child;
+  CNode * next;
+  int   start, end;
+};
+
+link_public_api(void)
+	string_delete(char *s);
+
+link_public_api(CNode *) 
+     linkage_constituent_tree(Linkage linkage);
+link_public_api(void)
+     linkage_free_constituent_tree(CNode * n);
+
 LINK_END_DECLS
 
 #endif
Only in link-grammar-4.2.2: link-grammar.spec
