Only in link-grammar-4.3.5.patched: bin
diff -ubr link-grammar-4.3.5/link-grammar/api.c link-grammar-4.3.5.patched/link-grammar/api.c
--- link-grammar-4.3.5/link-grammar/api.c	2008-04-21 08:51:29.000000000 -0700
+++ link-grammar-4.3.5.patched/link-grammar/api.c	2008-05-26 12:55:58.000000000 -0700
@@ -894,6 +894,10 @@
 	return linkage;
 }
 
+int linkage_get_current_sublinkage(Linkage linkage) {
+    return( linkage->current );
+}
+
 int linkage_set_current_sublinkage(Linkage linkage, int index) {
 	int errno;
 	if ((index < 0) ||
Only in link-grammar-4.3.5/link-grammar: link-features.h
diff -ubr link-grammar-4.3.5/link-grammar/link-grammar.def link-grammar-4.3.5.patched/link-grammar/link-grammar.def
--- link-grammar-4.3.5/link-grammar/link-grammar.def	2008-04-21 08:50:48.000000000 -0700
+++ link-grammar-4.3.5.patched/link-grammar/link-grammar.def	2008-05-26 12:54:55.000000000 -0700
@@ -75,6 +75,7 @@
 sentence_get_nth_word
 sentence_nth_word_has_disjunction
 linkage_create
+linkage_get_current_sublinkage
 linkage_set_current_sublinkage
 linkage_delete
 linkage_get_sentence
@@ -120,6 +121,7 @@
 linkage_constituent_node_get_end
 lperrno
 lperrmsg
+string_delete
 is_entity
 is_past_tense_form
 set_data_dir
diff -ubr link-grammar-4.3.5/link-grammar/link-includes.h link-grammar-4.3.5.patched/link-grammar/link-includes.h
--- link-grammar-4.3.5/link-grammar/link-includes.h	2008-04-21 08:59:29.000000000 -0700
+++ link-grammar-4.3.5.patched/link-grammar/link-includes.h	2008-05-26 12:51:22.000000000 -0700
@@ -223,6 +223,8 @@
 link_public_api(Linkage)
      linkage_create(int index, Sentence sent, Parse_Options opts);
 link_public_api(int)
+     linkage_get_current_sublinkage(Linkage linkage);
+link_public_api(int)
      linkage_set_current_sublinkage(Linkage linkage, int index);
 link_public_api(void)
      linkage_delete(Linkage linkage);
@@ -335,6 +337,10 @@
 extern link_public_api(char) 
      lperrmsg[];
 
+/* from utilities.c */
+link_public_api(void)
+	string_delete(char *s);
+
 LINK_END_DECLS
 
 #endif
diff -ubr link-grammar-4.3.5/link-grammar/read-dict.c link-grammar-4.3.5.patched/link-grammar/read-dict.c
--- link-grammar-4.3.5/link-grammar/read-dict.c	2008-04-01 09:47:17.000000000 -0700
+++ link-grammar-4.3.5.patched/link-grammar/read-dict.c	2008-05-26 12:53:28.000000000 -0700
@@ -702,14 +702,17 @@
 	} 
 	else if (is_idiom_word(dn->string))
 	{
+		if ( verbosity > 3 ) {
 		printf("*** Word \"%s\" found near line %d.\n", dn->string, dict->line_number);
 		printf("	Words ending \".Ix\" (x a number) are reserved for idioms.\n");
 		printf("	This word will be ignored.\n");
+		}
 		xfree((char *)dn, sizeof(Dict_node));
 	}
 	else if ((dn_head = abridged_lookup_list(dict, dn->string))!= NULL)
 	{
 		Dict_node *dnx;
+		if ( verbosity > 3 ) {
 		printf("*** The word \"%s\"", dn->string);
 		printf(" found near line %d of %s matches the following words:\n",
 			   dict->line_number, dict->name);
@@ -717,6 +720,7 @@
 			printf(" %s", dnx->string);
 		}
 		printf("\n	This word will be ignored.\n");
+		}
 		free_lookup_list(dn_head);
 		xfree((char *)dn, sizeof(Dict_node));
 	}
Only in link-grammar-4.3.5.patched: link-grammar-4.3.5.jar
Only in link-grammar-4.3.5: link-grammar.spec
